waffle-jest/src/matchers/calledOnContract/calledOnContractValidators.ts

```diff
@@ -0,0 +1,44 @@
+import {Contract} from 'ethers';
+import {MockProvider} from '@ethereum-waffle/provider';
+import {ProviderWithHistoryExpected} from './error';
+import {ensure} from './utils';
+
+export function validateContract(contract: any): asserts contract is Contract {
+  ensure(
+    contract instanceof Contract,
+    TypeError,
+    'argument must be a contract'
+  );
+}
+
+export function validateMockProvider(
+  provider: any
+): asserts provider is MockProvider {
```

Comment:
We could remove that assertion, lib could work with any provider that has callHistory (e.g. `buidler.dev` provider)

Reply:
Ahh, now it broke the build I Guess we could define type 
```ts 
interface ProviderWithHistory {
  callHistory ...
}
```
and keep the assertion with it instead of MockProvider

Reply:
where would I put this?

Reply:
In the same file I guess, it is just for validation purposes to let complier know about the type.
If it becomes problematic, just revert the commit and we can fix in in next version.